fortran/expand-interfaces.py

```diff
@@ -0,0 +1,234 @@
+#! /usr/bin/env python3
+
+from jinja2 import Environment
+
+class collectionCase():
```

Comment:
Please do not add a layer of code generation.

Fortran is a verbose language, and that is just something we have to live with. A code generation program benefits only the programmer who wrote it. For every other programmer of netcdf-fortran, it will be a barrier and confusing inconvenience. It will complexify the build system, and make debugging the code very difficult. It saves you some typing, one time, but add long-term costs and inconveniences. 

Instead, generate the code on your machine and commit only Fortran to netcdf-fortran.

Reply:
Well, you do generate the code with m4 and the macro-package is part of the netcdf-fortran package. I handle the issue the same way just with python.

Reply:
By the way, the code is already generated and provided and the result is provided in this pull-request.

Reply:
MOre striking to me is that the in 4.5.3 provided m4 genereator is not able to generate the provided code.

Reply:
Sorry about all the typos.

Reply:
OK, I belive the python code should be removed from this PR. The fact that you used it to generate the F90 code is for you alone to care about and maintain. ;-) The rest of the netcdf-fortran world only needs the Fortran code. It will not have to be generated again.

In terms of the m4, it's use is one of the most perennially annoying things in the world of netCDF. It is such a barrier to easy modification and debugging, that every time I found a bug with nc_test, I would first reproduce it in plain C code so I had a simple test I could work with. I highly question whether the fairly trivial amount of typing saved is worth all the extra time and effort that has gone into maintaining a bunch of m4 code in the netcdf-c build.

Reply:
I fully agree with your comment on m4 :-) However there are geeks which still understand it and even find it nice. There is a very funny idiom in German: I try to translate it into English. "You cannot debate about flavor!" said the monkey and bite into a piece of soap ...